trunk/SUAVE/Methods/Missions/Segments/Transition/Constant_Accelerated_Climb_Constant_Angle.py

```diff
@@ -0,0 +1,163 @@
+    N/A
+    """      
+    
+    # unpack
+    alt0        = segment.altitude_start 
+    altf        = segment.altitude_end 
+    climb_angle = segment.climb_angle
+    v0          = segment.air_speed_start_vector
+    vf          = segment.air_speed_end_vector
+    ax          = segment.acceleration   
+    T0          = segment.pitch_initial
+    Tf          = segment.pitch_final     
+    t_nondim    = segment.state.numerics.dimensionless.control_points
+    conditions  = segment.state.conditions 
+    
+    # check for initial and final altitude 
+    if alt0 is None:
+        if not segment.state.initials: raise AttributeError('altitude not set')
+        alt0 = -1.0 * segment.state.initials.conditions.frames.inertial.position_vector[-1,2]
+        segment.altitude_start  = alt0
+    
+    if altf is None:
+        raise AttributeError('final altitude not set')
+        
+    # check for initial pitch
+    if T0 is None:
+        T0  =  segment.state.initials.conditions.frames.body.inertial_rotations[-1,1]
+        segment.pitch_initial = T0    
+        
+    # check for initial velocity vector
+    if v0 is None:
+        v0  =  segment.state.initials.conditions.frames.inertial.velocity_vector[-1,:] 
+        segment.velocity_vector = v0  
+         
+    # discretize on altitude
+    alt = t_nondim * (altf-alt0) + alt0        
+        
+    # check for climb angle     
+    if climb_angle is None:
+        raise AttributeError('set climb')
+    
+    ground_distance = abs((altf-alt0)*np.arctan(climb_angle))
+    true_distance   = np.sqrt((altf-alt0)**2 + ground_distance**2)
+    t_initial       = conditions.frames.inertial.time[0,0]       
+    vf_mag          = np.linalg.norm(vf)
+    v0_mag          = np.linalg.norm(v0)
+    
+    if vf != None and ax != None:        
+        elapsed_time =  (-v0_mag + np.sqrt(v0_mag**2 + 4*0.5*ax*true_distance))/(2*ax) 
+        vf_mag = v0_mag + ax* (elapsed_time)
+    
+    if vf is None:
+        if ax is None: raise AttributeError('set either final speed or acceleration')
+        else:
+            elapsed_time =  (-v0_mag + np.sqrt(v0_mag**2 + 4*0.5*ax*true_distance))/(2*ax)
+            vf_mag = v0_mag + ax* (elapsed_time)
+        
+    if ax is None:
+        if vf is None: raise AttributeError('set either final speed or acceleration')
+        else:
+            ax = (vf_mag**2 - v0_mag**2)/(2*true_distance)
+            elapsed_time = (vf_mag - v0_mag)/ax     
+    
+    # dimensionalize time        
+    t_final   = t_initial + elapsed_time
+    t_nondim  = segment.state.numerics.dimensionless.control_points
+    time      = t_nondim * (t_final-t_initial)
+    
+    # Figure out vx
+    V  = (vf_mag-v0_mag) + v0_mag
+    vx = t_nondim *  V  * np.cos(climb_angle)
+    vz = t_nondim *  V  * np.sin(climb_angle)
+    
+    # set the body angle
+    body_angle =time*(Tf-T0)/(t_final-t_initial)
+    segment.state.conditions.frames.body.inertial_rotations[:,1] = body_angle[:,0]     
+
+    
```

Comment:
white space

Reply:
corrected, I needed one white space for aesthetics